src/Impl/RestPublisher.php

```diff
@@ -47,12 +48,29 @@ private function convertToPayloadTypeName($channelSchemaName): string
 
     public function send(Message $message): SuccessResult
     {
-        $json = $this->jsonProvider->getJson($message, $this->payloadTypeName);
-        $actualHeaders = $message->action == Message::PUBLISH_ACTION
-            ? array_merge($this->headers, ['Content-Type' => 'application/json; charset=UTF-8'])
-            : $this->headers;
+        $messageStatus = $this->sendMulti([$message])[0];
 
-        return $this->httpRequester->executePost($this->messageIngestionEndpointUri, $actualHeaders, $json);
+        if ($messageStatus->getSuccess() != null) {
+            return $messageStatus->getSuccess();
+        }
+
+        $failureResponse = $messageStatus->getFailure();
+        throw StreamxClientExceptionFactory::create(
+            $failureResponse->getErrorCode(),
+            $failureResponse->getErrorMessage()
+        );
+    }
+
+    public function sendMulti(array $messages): array
+    {
+        $messageJsons = [];
+        foreach ($messages as $message) {
+            $messageJsons[] = $this->jsonProvider->getJson($message, $this->payloadTypeName);
+        }
+        $multiMessageJson = implode("\n", $messageJsons);
```

Comment:
Comparing to the draft version of this PR - nuw the user doesn't have to take care of:
 - injecting the payload type name
 - serializing the messages to json
 - joining the messages

Reply:
OK, I'll try that.
But maybe even for some debugging reasons in the future, we would want to have the line breaks, to not receive one huge line?